agent/config/config.go

```diff
@@ -101,6 +109,7 @@ type Config struct {
 	BindAddr                         *string                  `json:"bind_addr,omitempty" hcl:"bind_addr" mapstructure:"bind_addr"`
 	Bootstrap                        *bool                    `json:"bootstrap,omitempty" hcl:"bootstrap" mapstructure:"bootstrap"`
 	BootstrapExpect                  *int                     `json:"bootstrap_expect,omitempty" hcl:"bootstrap_expect" mapstructure:"bootstrap_expect"`
+	Cache                            Cache                    `json:"cache,omitempty" hcl:"cache" mapstructure:"cache"`
```

Comment:
The rpc equivalent is in `limits`. Since there is no existing `cache` config just yet, I am wondering if putting it into `limits` is better. If so I would lean towards naming the options `cache_limit` and `cache_max_burst`.

Reply:
I don't know, I personally like to have structured data (and I initially planned to have more complex cache configuration), hence the dedicated section, but this could be changed. Tell me if you want me to change it.

@dnephin do you have an opinion?

Reply:
@i0rek There is a downside to move it into limits: {}

=> now cache {} is consistent between Runtime and Configuration (aka same name & so on).
@dnephin asked me to use `cache.Config` in the runtime (so we do not duplicate and use directly in constructor of Cache)

So we have:

Config.Cache.entry_fetch* to configure
And
Runtime.Cache.entry* to see applied configuration (this is consistent, and shown properly in /v1/agent/self


If I move to limits, I'll still have to keep Cache in Runtime because it is actually a `cache.Options` as requested by @dnephin 

The good thing about cache is Consistency (which is cool for a cache 🥇 ) between Runtime and Config.
Furthermore, limits are currently dispatched at various places in Runtime as flat... (I personally hate that Runtime as shown in /v1/agent/self and config have completely different layouts and names, this is hard to compare actual values in server and config, but there are probably too many people relying on those names)

Reply:
I think a cache section makes more sense. I like the idea of grouping by components or sub-systems, I think that is a more natural way to organize them.

In the future we might want to make it `api.cache`, but I think that would be better as a larger change to more config fields.

Reply:
@dnephin I feel the same (and would like to avoid changing it one more time :-) - But if you decide otherwise, I'll do it